src/lib.rs

```diff
@@ -146,14 +146,17 @@ where
     /// An execution error.
     type Error;
 
+    /// A `SeedableRng`, suitable to generate randomness.
+    type Rng: SeedableRng;
+
     /// Execute an algorithm over a context and graph.
     /// Changes to the context will be persisted across
     /// executions of the algorithm.
-    fn execute<R: SeedableRng>(
+    fn execute(
         &self,
         context: &mut Self::Context,
         graph: &mut G,
-        rng: R,
+        rng_initial_seed: <Self::Rng as SeedableRng>::Seed,
```

Comment:
The thing that's weird I think is that we define an Rng type, but only use the Seed of it. Perhaps we don't need the Rng at all? Just pass in some kind of seed?

Reply:
We could have the associated type be simply `type Rng` and forgo the `: SeedableRng` constraint. This will also remove the `rand` dependency from the crate. However, this begs the question if this is what we want; on the contrary of the `type Id: Clone` discussion we had yesterday, here we are stipulating an undeniable contract between this trait and the implementor, which is that `Rng` _must be_ a `SeedableRng`, which I would have thought it's a good thing, because it restrict the degree of freedom in implementing the trait.

Thoughts?

Reply:
Yeah, but we don't actually need an RNG here. We just need a random seed - so anything that can be converted to `Seed` should do, ie. a `[u8]`

Reply:
Basically, osrank can choose whatever RNG it wants, all we provide is entropy.